### PR TITLE
Update pom file version scanner

### PR DESCRIPTION
**Pull request by JimSuplizio**
The primary reason for this PR was that I'd found, by happenstance, a number of issues where dependency tags, weren't correctly set.

The pom_file_version_scanner now figures out, by using the ci*.yml files, what builds in each sdk/<ServiceDirectory> and verifies that any tags appropriately. To quickly reiterate the rules:

1. Any libraries that build and release in the same sdk/<ServiceDirectory> **must** be using the current versions of other libraries that build and release in the same ServiceDirectory. There are no exceptions for this. If a library doesn't meet this criteria, then it needs to be in another pipeline, it's that simple.
2. AdditionalModules in a ci*.yml file is used to identify libraries that are needed, usually for testing, that are not released. These are part of the module list used to determine sparse-checkout directories. 
3. Tags for these AdditionalModules in pom.xml files must be _current_. The reason for this is that these are always built from source but, in the case of an accidental release, it ensures that nothing is stuck using an older, possibly defunct version.

For most service directories, the updates consisted of adding the correct additional modules and/or correctly setting tags. There were two special cases that required new pipelines to be created. The reason being is that these need to use the dependency versions of the libraries in their respective service directories because they're not released with the same cadence and have different owners than the service directory owner. As such, the ci.yml file for their pipelines was created in the library's subdirectory. We already do this in sdk/communication since each library is owned by a different team.

1. azure-resourcemanager-eventgrid
2. azure-identity-extensions

### CODE REVIEW

File: eng/versioning/pom_file_version_scanner.ps1 [hunk 1]
```diff
@@ -782,12 +895,12 @@ Get-ChildItem -Path $Path -Filter pom*.xml -Recurse -File | ForEach-Object {
...
-    $ElapsedTime = $(get-date) - $StartTime
-    $TotalRunTime = "{0:HH:mm:ss}" -f ([datetime]$ElapsedTime.Ticks)
-    Write-Host "Total run time=$($TotalRunTime)"
-}
+# if ($UseVerboseLogging)
+# {
+$ElapsedTime = $(get-date) - $StartTime
+$TotalRunTime = "{0:HH:mm:ss}" -f ([datetime]$ElapsedTime.Ticks)
+Write-Host "Total run time=$($TotalRunTime)"
+# }
```
Comment (billwert) [inline eng/versioning/pom_file_version_scanner.ps1:903]: nit: Either uncomment or delete the lines.

File: sdk/identity/ci.yml [hunk 2]
```diff
@@ -40,10 +40,6 @@ parameters:
   displayName: 'azure-identity'
   type: boolean
   default: true
-- name: release_azureidentityextensions
```
Comment (billwert) [inline sdk/identity/ci.yml:43]: should azure-identity-extensions be added to the exclude list above as was done in the EG case?

**Review by billwert — APPROVED** [review]
Couple questions but otherwise looks good to me.

File: sdk/identity/ci.yml (hunk 2 above)
Comment (JimSuplizio) [reply sdk/identity/ci.yml:43]: Good catch @billwert. Next commit will add `- sdk/identity/azure-identity-extensions/` to both the Trigger and PR exclude lists.

**Review by alzimmermsft — APPROVED** [review]

**Review by alzimmermsft — APPROVED** [review]

**Comment by JimSuplizio** [comment]
I'm going to override check-enforcer. The reason being is that the latest-jdk.yml had to be updated for the generate_from_source_pom argument changes. If these changes were going to cause a failure, it would have happened long before the run tests step even started and latest-jdk.yml takes 2 hours to run.

**Pull request merged**